zaza/openstack/charm_tests/ceilometer/tests.py

```diff
@@ -0,0 +1,118 @@
+
+    def test_900_restart_on_config_change(self):
+        """Checking restart happens on config change."""
+        # Expected default and alternate values
+        current_value = openstack_utils.get_application_config_option(
+            'ceilometer', 'debug'
+        )
+        assert type(current_value) == bool
+        new_value = not current_value
+
+        # Convert bool to str
+        current_value = str(current_value)
+        new_value = str(new_value)
+
+        set_default = {'debug': current_value}
+        set_alternate = {'debug': new_value}
+        default_entry = {'DEFAULT': {'debug': [current_value]}}
+        alternate_entry = {'DEFAULT': {'debug': [new_value]}}
+
+        # Config file affected by juju set config change
+        conf_file = '/etc/ceilometer/ceilometer.conf'
+        services = {}
+        current_release = openstack_utils.get_os_release()
+        xenial_pike = openstack_utils.get_os_release('xenial_pike')
+        xenial_ocata = openstack_utils.get_os_release('xenial_ocata')
+        xenial_newton = openstack_utils.get_os_release('xenial_newton')
+        trusty_mitaka = openstack_utils.get_os_release('trusty_mitaka')
+        trusty_liberty = openstack_utils.get_os_release('trusty_liberty')
+
+        if current_release >= xenial_pike:
+            services = {
+                'ceilometer-polling: AgentManager worker(0)': conf_file,
+                'ceilometer-agent-notification: NotificationService worker(0)':
+                    conf_file,
+            }
+        elif current_release >= xenial_ocata:
+            services = {
+                'ceilometer-collector: CollectorService worker(0)': conf_file,
+                'ceilometer-polling: AgentManager worker(0)': conf_file,
+                'ceilometer-agent-notification: NotificationService worker(0)':
+                    conf_file,
+                'apache2': conf_file,
+            }
+        elif current_release >= xenial_newton:
+            services = {
+                'ceilometer-collector - CollectorService(0)': conf_file,
+                'ceilometer-polling - AgentManager(0)': conf_file,
+                'ceilometer-agent-notification - NotificationService(0)':
+                    conf_file,
+                'ceilometer-api': conf_file,
+            }
+        else:
+            services = {
+                'ceilometer-collector': conf_file,
+                'ceilometer-api': conf_file,
+                'ceilometer-agent-notification': conf_file,
+            }
+
+            if current_release < trusty_mitaka:
+                services['ceilometer-alarm-notifier'] = conf_file
+                services['ceilometer-alarm-evaluator'] = conf_file
+
+            if current_release >= trusty_liberty:
+                # Liberty and later
+                services['ceilometer-polling'] = conf_file
+            else:
+                # Juno and earlier
+                services['ceilometer-agent-central'] = conf_file
```

Comment:
Please move the retrieval of the services list to a class method (or property) so that other tests can use it (see [cinder tests](https://github.com/openstack-charmers/zaza-openstack-tests/blob/master/zaza/openstack/charm_tests/cinder/tests.py#L140)